lib/media/streaming_engine.js

```diff
@@ -1519,9 +1519,11 @@ shaka.media.StreamingEngine = class {
           logPrefix, 'looking up segment from new stream endTime:', time);
 
       const reverse = this.playerInterface_.getPlaybackRate() < 0;
-      mediaState.segmentIterator =
-          mediaState.stream.segmentIndex.getIteratorForTime(
-              time, /* allowNonIndepedent= */ false, reverse);
+      if (mediaState.stream.segmentIndex) {
```

Comment:
If this is null, is segmentIterator always null already?

Reply:
Yes!